client/me/concierge/main.js

```diff
@@ -88,9 +96,24 @@ export class ConciergeMain extends Component {
 		if ( ! availableTimes || ! site || ! site.plan || null == scheduleId || ! userSettings ) {
 			return <Skeleton />;
 		}
+		let hasBusinessOnboardingExpired;
+		if ( currentPlan ) {
+			const expiryDateMoment = this.props.moment( currentPlan.expiryDate );
+			const is2YearPlan = currentPlan.productSlug === PLAN_BUSINESS_2_YEARS;
+			const businessOnboardingExpiration = this.props.moment(
+				is2YearPlan ? PLAN_BUSINESS_2Y_ONBOARDING_EXPIRE : PLAN_BUSINESS_ONBOARDING_EXPIRE
+			);
+
+			hasBusinessOnboardingExpired = businessOnboardingExpiration.diff( expiryDateMoment ) < 0;
+		}
+
+		const hasIncludedSessions = scheduleId === 1;
+		const hasPurchasedSessions = scheduleId > 1;
 
-		// if scheduleId is 0, it means the user is not eligible for the concierge service.
-		if ( scheduleId === 0 ) {
+		const isBusinessOnboardingAvailable =
+			hasPurchasedSessions || ( hasIncludedSessions && ! hasBusinessOnboardingExpired );
+
+		if ( ! isBusinessOnboardingAvailable ) {
```

Comment:
Nitpick: I think the variable name `isBusinessOnboardingAvailable ` could be different, like maybe `hasAvailableSessions`.  AFAIK "Business onboarding" was used to refer to the free included sessions we offered to Business plan users, and the variable seems to indicate that we show the upsell only to those who don't have business onboarding, but we show it even to those who purchased a Quick Start and then used it up etc.

Reply:
That makes sense. Changed to `hasAvailableSessions`.